k8s/client.go

```diff
@@ -35,13 +36,22 @@ type ClientConfig struct {
 	CustomMetadataIsAnyType bool
 
 	MetricsConfig metrics.Config
+
+	// NegotiatedSerializerProvider is a function which provides a runtime.NegotiatedSerializer for the underlying
+	// kubernetes rest.RESTClient, if defined.
+	NegotiatedSerializerProvider func(kind resource.Kind) runtime.NegotiatedSerializer
 }
 
 // DefaultClientConfig returns a ClientConfig using defaults that assume you have used the SDK codegen tooling
 func DefaultClientConfig() ClientConfig {
 	return ClientConfig{
 		CustomMetadataIsAnyType: false,
 		MetricsConfig:           metrics.DefaultConfig(""),
+		/* NegotiatedSerializerProvider: func(kind resource.Kind) runtime.NegotiatedSerializer {
```

Comment:
Not putting this as the default out of an overabundance of caution in changing too much at once. Introduced this as initially it seemed to be needed by the new informer, but as long as we leave the example object `nil` (like in `KubernetesBasedInformer`), we can continue to use the generic negotiated serializer.